File: Scripts/HealthPotionButton.cs
using UnityEngine;
using System.Collections;

public class HealthPotionButton : MonoBehaviour {
	private GameObject player;
	private Health playerHealthObj;
	private Globals globals;
	public GameObject healthPotionsLabelGO;
	private UILabel healthPotionsLabel;
	
	void Awake()
	{
		globals = Globals.GetInstance();
	
		player = GameObject.FindWithTag("Player");
		playerHealthObj = player.GetComponent<Health>();
		
		healthPotionsLabel = healthPotionsLabelGO.GetComponent<UILabel>();
		
		healthPotionsLabel.text = globals.healthPotionsNumber + "";
		
		GetComponent<UIButtonSound>().volume = 0;
	}
	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
	
	}
	
	void OnClick()
	{
		if(globals.healthPotionsNumber > 0 && playerHealthObj.health < playerHealthObj.maxHealth)
		{
			GetComponent<UIButtonSound>().volume = 1;
			Debug.Log("globals.healthPotionsNumber " + globals.healthPotionsNumber);
			globals.healthPotionsNumber -= 1;
			PlayerPrefs.SetInt("healthPotionsNumber", globals.healthPotionsNumber);
			healthPotionsLabel.text = globals.healthPotionsNumber + "";
			playerHealthObj.health = playerHealthObj.maxHealth;
		}
		else
		{
			GetComponent<UIButtonSound>().volume = 0;	
		}
		
		
	}
}

File: Scripts/HealthPotionButton.js
#pragma strict
private var player : GameObject;
private var playerHealthObj : Health;
private var healthPotionsNumber : int;
private var globals : Globals;
//public var healthPotionsLabel : UILabel;
function Awake()
{
	globals = Globals.GetInstance();
	
	healthPotionsNumber = globals.healthPotionsNumber;
	
	player = GameObject.FindWithTag("Player");
	playerHealthObj = player.GetComponent.<Health>();
}

function OnClick()
{
	if(healthPotionsNumber > 0)
	{
		globals.healthPotionsNumber -= 1;
		PlayerPrefs.SetInt("healthPotionsNumber", globals.healthPotionsNumber);
		playerHealthObj.health = playerHealthObj.maxHealth;
	}
	
	
}